src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1317,9 +1317,9 @@ def __init__(self,
                  state_update_notify: Optional[bool] = True,
                  state_update_interval_sec: Optional[int] = 5,
                  enable_spdk_discovery_controller: Optional[bool] = False,
-                 omap_file_lock_duration: Optional[int] = 60,
-                 omap_file_lock_retries: Optional[int] = 15,
-                 omap_file_lock_retry_sleep_interval: Optional[int] = 5,
+                 omap_file_lock_duration: Optional[int] = 20,
+                 omap_file_lock_retries: Optional[int] = 30,
+                 omap_file_lock_retry_sleep_interval: Optional[float] = 1.0,
```

Comment:
@VallariAg pointed me at this PR after I brought up a test failure caused by `ValueError` exception in an integration branch run based on squid:
```
2024-05-11T22:28:06.968 INFO:journalctl@ceph.nvmeof.nvmeof.a.smithi018.stdout:May 11 22:28:06 smithi018 ceph-cd58ef28-0fe4-11ef-bc98-c7b262605968-nvmeof-nvmeof-a[54398]:   File "/src/control/state.py", line 184, in __init__
2024-05-11T22:28:06.968 INFO:journalctl@ceph.nvmeof.nvmeof.a.smithi018.stdout:May 11 22:28:06 smithi018 ceph-cd58ef28-0fe4-11ef-bc98-c7b262605968-nvmeof-nvmeof-a[54398]:     self.omap_file_lock_retry_sleep_interval = self.omap_state.config.getint_with_default("gateway",
2024-05-11T22:28:06.968 INFO:journalctl@ceph.nvmeof.nvmeof.a.smithi018.stdout:May 11 22:28:06 smithi018 ceph-cd58ef28-0fe4-11ef-bc98-c7b262605968-nvmeof-nvmeof-a[54398]:   File "/src/control/config.py", line 47, in getint_with_default
2024-05-11T22:28:06.969 INFO:journalctl@ceph.nvmeof.nvmeof.a.smithi018.stdout:May 11 22:28:06 smithi018 ceph-cd58ef28-0fe4-11ef-bc98-c7b262605968-nvmeof-nvmeof-a[54398]:     return self.config.getint(section, param, fallback=value)
2024-05-11T22:28:06.969 INFO:journalctl@ceph.nvmeof.nvmeof.a.smithi018.stdout:May 11 22:28:06 smithi018 ceph-cd58ef28-0fe4-11ef-bc98-c7b262605968-nvmeof-nvmeof-a[54398]:   File "/usr/lib64/python3.9/configparser.py", line 818, in getint
2024-05-11T22:28:06.969 INFO:journalctl@ceph.nvmeof.nvmeof.a.smithi018.stdout:May 11 22:28:06 smithi018 ceph-cd58ef28-0fe4-11ef-bc98-c7b262605968-nvmeof-nvmeof-a[54398]:     return self._get_conv(section, option, int, raw=raw, vars=vars,
2024-05-11T22:28:06.969 INFO:journalctl@ceph.nvmeof.nvmeof.a.smithi018.stdout:May 11 22:28:06 smithi018 ceph-cd58ef28-0fe4-11ef-bc98-c7b262605968-nvmeof-nvmeof-a[54398]:   File "/usr/lib64/python3.9/configparser.py", line 808, in _get_conv
2024-05-11T22:28:06.969 INFO:journalctl@ceph.nvmeof.nvmeof.a.smithi018.stdout:May 11 22:28:06 smithi018 ceph-cd58ef28-0fe4-11ef-bc98-c7b262605968-nvmeof-nvmeof-a[54398]:     return self._get(section, conv, option, raw=raw, vars=vars,
2024-05-11T22:28:06.969 INFO:journalctl@ceph.nvmeof.nvmeof.a.smithi018.stdout:May 11 22:28:06 smithi018 ceph-cd58ef28-0fe4-11ef-bc98-c7b262605968-nvmeof-nvmeof-a[54398]:   File "/usr/lib64/python3.9/configparser.py", line 803, in _get
2024-05-11T22:28:06.969 INFO:journalctl@ceph.nvmeof.nvmeof.a.smithi018.stdout:May 11 22:28:06 smithi018 ceph-cd58ef28-0fe4-11ef-bc98-c7b262605968-nvmeof-nvmeof-a[54398]:     return conv(self.get(section, option, **kwargs))
2024-05-11T22:28:06.969 INFO:journalctl@ceph.nvmeof.nvmeof.a.smithi018.stdout:May 11 22:28:06 smithi018 ceph-cd58ef28-0fe4-11ef-bc98-c7b262605968-nvmeof-nvmeof-a[54398]: ValueError: invalid literal for int() with base 10: '1.0'
```
https://pulpito.ceph.com/teuthology-2024-05-11_21:16:03-rbd-squid-distro-default-smithi/7702717

Apparently there was a mismatch in the type expected by the NVMeOF container version that got pulled in that run and the float set by cephadm here (see https://github.com/ceph/ceph-nvmeof/pull/597).  This highlights that maintaining two sets of defaults, one in the gateway itself and one in cephadm is fragile.  I would suggest dropping the defaults for options that have respective defaults in the gateway here.

Reply:
we have https://tracker.ceph.com/issues/65718 for that same issue from the orch side, and have fixed it in main. I'll get the squid backport open soon and will make sure to test in orch and the rbd:nvmeof job you mentioned below

Reply:
> I would suggest dropping the defaults for options that have respective defaults in the gateway here.

@adk3798 What is your take on this?  I added it to the agenda for next week's NVMeOF meeting [1], but want to make sure there isn't a strong reason for `NvmeofServiceSpec` to provide a default for everything first.

[1] https://pad.ceph.com/p/rbd_nvmeof

Reply:
I actually wasn't aware the nvmeof daemon itself had defaults for these options until today, but knowing that I would definitely support just using those instead of sticking the defaults into the service spec